packages/next/src/lib/metadata/resolve-title.ts

```diff
@@ -0,0 +1,30 @@
+import type { Metadata } from './types/metadata-interface'
+import type { AbsoluteTemplateString } from './types/metadata-types'
+
+function resolveTitleTemplate(template: string | null, title: string) {
+  return template ? template.replace(/%s/g, title) : title
+}
+
+export function resolveTitle(
+  stashedTemplate: string | null,
+  title: Metadata['title']
+) {
+  const resolved: AbsoluteTemplateString = {
+    absolute: '',
+    template: null,
+  }
```

Comment:
I'm sorry I'm being a stickler here but I really think we should avoid as many allocations as possible because the metadata process will be invoked on every request and it blocks the shell from streaming and while it probably won't be slower than your app code we don't want it to compete for resources.

If you think of `ResolvedMetadata` as a mutation target this function should accept the mutable AbsoluteTemplateString from `ResolvedMetadata` and mutate it rather than returning a new object.

maybe a better name is `function mergeTitle` so it's clear the function itself is going merge into some mutable target value rather than returning some resolved value that will get assigned later.

Is there a reason you don't want to set up the code this way? I may be missing something but if I'm not I'd like to make this as efficient as possible